gst/tensor_merge/gsttensormerge.c

```diff
@@ -371,52 +372,102 @@ gst_merge_tensors_config (GstTensorMerge * tensor_merge,
     GstTensorsConfig * configs, GstTensorConfig * config)
 {
   gboolean ret = FALSE;
+  int i, first, second, third, fourth;
```

Comment:
If you simply use tensor_dim along with RANK_LIMIT, you can make the code general and more readable.

Reply:
Right. Done!